### ISSUE TITLE
[BUG]national players names are not displayed

### ISSUE THREAD

**Issue by wsbrenk** [akasolace/HO#1025]
Feedback from Hattrick forum:

hi
I'm the new BC for Surinam and i'm trying to get the National Team in HO

I managed to get the players in but I don't see there names (I do see there skills)

wich would make it difficult to do anything

how do I make it so that I see there names also?

tnx
Stan

**Comment by wsbrenk** [comment]
@stan i think i found the error. 

i see 2 ways testing it:
 *  if you trust me, you could upload your HO-db of your national team and i will use it to debug.
 * otherwise i create a pull request and @akasolace could include it in the next dev-version of HO5. Then we need you to test my changes.

**Comment by wsbrenk** [comment]
@akasolace didn't get the database until now - so no chance to debug it. Please verify my pull request and we should ask the trainer to test it.

**Comment by wsbrenk** [comment]
same as #629

**Comment by StanMarsh1** [comment]
he he
finally I got in

my situation looks exactly like the one you just discribed

so far I've done the following

I set up the NT in HO
tried to download but nothing happens
because I saw the hfr I thought to try and import wich works for the half (I don't see the names and I can't make a lineup)

now i'm even unable to download 
even after removing the second account nothing happens
I tried to remove all hrf files and start over but nothing happens

how do I upload my HO db from the national team?

**Comment by akasolace** [comment]
should be fixed by #1028

**Comment by StanMarsh1** [comment]
[db_BC-2021-05-10.zip](https://github.com/akasolace/HO/files/6487367/db_BC-2021-05-10.zip)
[db_BC-2021-05-11.zip](https://github.com/akasolace/HO/files/6487368/db_BC-2021-05-11.zip)
[db_BC-2021-05-12.zip](https://github.com/akasolace/HO/files/6487369/db_BC-2021-05-12.zip)
[db_BC-2021-05-14.zip](https://github.com/akasolace/HO/files/6487376/db_BC-2021-05-14.zip)
[db_National Team-2021-05-14.zip](https://github.com/akasolace/HO/files/6487377/db_National.Team-2021-05-14.zip)
[db_user2-2021-05-14.zip](https://github.com/akasolace/HO/files/6487378/db_user2-2021-05-14.zip)
[db_StanMarsh1-2021-05-12.zip](https://github.com/akasolace/HO/files/6487379/db_StanMarsh1-2021-05-12.zip)
[db_StanMarsh1-2021-05-13.zip](https://github.com/akasolace/HO/files/6487380/db_StanMarsh1-2021-05-13.zip)
[db_StanMarsh1-2021-05-14.zip](https://github.com/akasolace/HO/files/6487381/db_StanMarsh1-2021-05-14.zip)

**Comment by StanMarsh1** [comment]
hi
sorry it took a bit longer then I wanted
first the registraton on this site took a long time
then for some reason windows doesn't let me see "appdata" 
and the indexing wasn't set right which makes searching difficult

any way I hope these are the files yu are looking for
the first 3 are off course from the national team
the second three also (I was trying some things)
the last three are of my team

thanks 
Stan

**Comment by StanMarsh1** [comment]
huh
I'm trying to upload a new zip file but I'm unable to attach a file

**Comment by StanMarsh1** [comment]
[db_BC-2021-05-15.zip](https://github.com/akasolace/HO/files/6488372/db_BC-2021-05-15.zip)

**Comment by StanMarsh1** [comment]
succes lol

so I made a new name so that I had to get authorization again
that's why there is only one zip
I'm assuming it makes a new one next day

**Comment by StanMarsh1** [comment]
i'm trying to attach the files so far no succes

not that the are any bigger the two I have right now are 19 kb in size

**Comment by StanMarsh1** [comment]
[db_BC-2021-05-15.zip](https://github.com/akasolace/HO/files/6489377/db_BC-2021-05-15.zip)
[db_BC-2021-05-16.zip](https://github.com/akasolace/HO/files/6489378/db_BC-2021-05-16.zip)

**Comment by wsbrenk** [comment]
names of players are downloaded and stored in an hrf file.
 hrf file is not imported automatically.
 I hope i can change that too.

**Comment by akasolace** [comment]
@wsbrenk is this one supposed to be fixed or does it requires more work ?

**Comment by wsbrenk** [comment]
@akasolace the workflow download nt data to hrf file + import hrf file separatly is fixed. I find this too cumbersome and i don't like it so much that it only works that way at the moment.

i will try to refresh player overview automatically and would be glad if someone could support, since i'm not sure that this manual import of nt-hrf was the intended workflow. 

Progress will be slow down the next days, my holidays with bad weather at home are over.

**Comment by akasolace** [comment]
I will need some clarification as I am not sure I follow, 
my understanding was that for both NT and normal database, the workflow is: 
  `data download -> hrf creation -> model creation from hrf`

Is it not the case for NT database or do you want to change the workflow ?

**Comment by wsbrenk** [comment]
@akasolace your correct: this workflows runs automatically in case of our normal teams
in case of nt teams i had to import the created hrf with FILE->Import hrf file.

**Comment by akasolace** [comment]
ah ok this is strange ... is it intentional or is it a bug ?
I dont even see where in the code the distinction is made between NT and non NT regarding this ...

**Comment by wsbrenk** [comment]
@akasolace if (UserManager.instance().getCurrentUser().isNtTeam()) {

**Comment by akasolace** [comment]
Ok thank you

**Comment by wsbrenk** [comment]
![image](https://user-images.githubusercontent.com/11821502/119028390-af7fd700-b9a7-11eb-87ad-a871ae393153.png)

Download worked for the first time - view is refreshed automatically showing no players anymore, which is correct since Stan removed all his NT players.
The refresh took very long. I think there are things in MatchAndLineupSelectionPanel that can not handle a nt team refresh.

**Comment by wsbrenk** [comment]
TODOs:
* display of nt download info (text area) is not displayed
* check performance of view refresh after nt download

**Comment by StanMarsh1** [comment]
hi
so today i put back 26 players and I wanted to download them to see what I had to play with
but to my surprise I wasn't able to download
Do I need to make a completly new team or....

**Comment by wsbrenk** [comment]
@StanMarsh1 as i wrote in ht-mail. please wait until @akasolace created the next dev version of HO5 and then try again to download your new team data. since ho5 needs new chpp-authorization i then need again your nt database if i should further debug with your team data. Thanks

**Comment by akasolace** [comment]
New dev artefacts have been produced yesterday.
@wsbrenk upgrade from 4.0 does not work for NT database ?

**Comment by wsbrenk** [comment]
@akasolace upgrade should work same way as for normal teams. first download with new version will ask for a new authorization of HO, since the new scope "manage_youth" is added.

**Comment by akasolace** [comment]
@wsbrenk ok I will release new artefacts in the weekend. I will try to complety finalize #814 by then

**Comment by wsbrenk** [comment]
new players with names are displayed.

* training panel gives npe
* matches are not shown (downloaded)

**Comment by akasolace** [comment]
Sounds all broken for national teams ...

**Comment by wsbrenk** [comment]
@StanMarsh1 @akasolace i think 
* prio 1 should be matches download
* prio 2 team analyzer (opponent matches download)
do you agree?

i see no chances to calc players training effects (subskills) with reasonable effort.

**Comment by akasolace** [comment]
@wsbrenk I agree, I don't think training is very relevant for NT teams. I think we should indeed focus on match download and team analyzer. I am not sure of the work required, if it is small we can put this for 5.0 otherwise we put this for 6.0.

**Comment by akasolace** [comment]
@wsbrenk also more generally, I think that all new features should go automatically to milestone 6.0 but "bugs" ticket will go in for 5.0.  I will let you create the tickets and we decide afterward.  But meanwhile, is this ticket #1025 closed ?

**Comment by wsbrenk** [comment]
@akasolace i will create 2 new tickets concerning NT features. so this one could be closed.
